tooling/src/hypothesistooling/projects/hypothesispython.py

```diff
@@ -0,0 +1,242 @@
+            'line of the file should be RELEASE_TYPE: followed by one of '
+            'major, minor, or patch, to specify the type of release that '
+            'this is (i.e. which version number to increment). Instead the '
+            'first line was %r' % (release_lines[0],)
+        )
+        sys.exit(1)
+
+    return release_type, release_contents
+
+
+def update_changelog_and_version():
+    global __version_info__
+    global __version__
+
+    with open(CHANGELOG_FILE) as i:
+        contents = i.read()
+    assert '\r' not in contents
+    lines = contents.split('\n')
+    assert contents == '\n'.join(lines)
+    for i, l in enumerate(lines):
+        if CHANGELOG_ANCHOR.match(l):
+            assert CHANGELOG_BORDER.match(lines[i + 2]), repr(lines[i + 2])
+            assert CHANGELOG_HEADER.match(lines[i + 3]), repr(lines[i + 3])
+            assert CHANGELOG_BORDER.match(lines[i + 4]), repr(lines[i + 4])
+            beginning = '\n'.join(lines[:i])
+            rest = '\n'.join(lines[i:])
+            assert '\n'.join((beginning, rest)) == contents
+            break
+
+    release_type, release_contents = parse_release_file()
+
+    new_version = list(__version_info__)
+    bump = VALID_RELEASE_TYPES.index(release_type)
+    new_version[bump] += 1
+    for i in range(bump + 1, len(new_version)):
+        new_version[i] = 0
+    new_version = tuple(new_version)
+    new_version_string = '.'.join(map(str, new_version))
+
+    __version_info__ = new_version
+    __version__ = new_version_string
+
+    with open(VERSION_FILE) as i:
+        version_lines = i.read().split('\n')
+
+    for i, l in enumerate(version_lines):
+        if 'version_info' in l:
+            version_lines[i] = '__version_info__ = %r' % (new_version,)
+            break
+
+    with open(VERSION_FILE, 'w') as o:
+        o.write('\n'.join(version_lines))
+
+    now = datetime.utcnow()
+
+    date = max([
+        d.strftime('%Y-%m-%d') for d in (now, now + timedelta(hours=1))
+    ])
+
+    heading_for_new_version = ' - '.join((new_version_string, date))
+    border_for_new_version = '-' * len(heading_for_new_version)
+
+    new_changelog_parts = [
+        beginning.strip(),
+        '',
+        '.. _v%s:' % (new_version_string),
+        '',
+        border_for_new_version,
+        heading_for_new_version,
+        border_for_new_version,
+        '',
+        release_contents,
+        '',
+        rest
+    ]
+
+    with open(CHANGELOG_FILE, 'w') as o:
+        o.write('\n'.join(new_changelog_parts))
+
+
+def commit_pending_release():
+    git('rm', RELEASE_FILE)
+    git('add', CHANGELOG_FILE, VERSION_FILE)
+
+    git(
+        'commit', '-m',
+        'Bump version to %s and update changelog\n\n[skip ci]' % (__version__,)
+    )
+
+
+def update_changelog_for_docs():
+    if not tools.has_release():
+        return
+    if tools.has_uncommitted_changes(tools.CHANGELOG_FILE):
```

Comment:
Why is this `tools.CHANGELOG_FILE`? I don’t see it in `hypothesistooling.py` any more.

Reply:
It shouldn't be. Hmm I wonder if this code is ever called now.